### ISSUE TITLE
Move indicator calculators into their own packages

### ISSUE THREAD

**Issue by balupton** [rrag/react-stockcharts#306]
Hi there,

I've been looking into automated trading, and specifically the renko indicator. I've noticed that your package is pretty darn awesome, and includes renko.

Currently, the indicator calculators are bundled with react-stockcharts and intended to be used with it.  Looking at their source code, their doesn't seem to be anything specific in them that requires the client-side. As such, it would be cool for them to published as independent and documented universal npm packages, such that they can run on the client-side in web browsers and server-side in nodejs.

That will then allow stockcharts, as well as node apps, to use your calculator code easily, increasing opportunity for use and improvement.

What do you think?

**Comment by balupton** [comment]
If you are down, I've created https://github.com/stockcharts and invited you as an owner.

I could work on abstracting the calculators out, and submit PRs to the `react-stockcharts` repo for each abstraction.

**Comment by rrag** [comment]
you are right, there is nothing react specific. if you notice the usage also it is like

```
import { sar } from "react-stockcharts/lib/indicator";
```

I was planing to use lerna and create a monorepo with mutiple packages. However as you might have seen there are 0 tests in the project so far, I am not comfortable publishing these packages without sufficient test. I am down with the new org, and I am happy to contribute and would appreciate support to creating tests also.

oh and I almost forgot, the indicators in react-stockcharts are inspired by [d3fc](https://github.com/d3fc/d3fc-technical-indicator)

**Comment by balupton** [comment]
Cool.

I've also noticed https://github.com/anandanand84/technicalindicators - maybe this task could be simplified by using their indicators instead?

**Comment by rrag** [comment]
I am not familiar with that lib, I will have to check it out. When starting down the custom indicator route, I did look at using an existing library, it has its benefits, but also might lack customization.

for instance, I wanted to be able to customize the type of moving average on a bollingerband, the default is `sma`, and the libraries I was looking at did not allow such, few of them required data to be formatted as an array of prices, while I already have data as an array of `ohlcv`.and I did not want to do an additional map operation. Finally I really liked the style in d3fc, using functional constructs, minimizing object mutation so I took the plunge.

Still, renko, p&f, kagi are not written d3fc style, I will have to get to it.
